shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/user/ShardingSphereUser.java

```diff
@@ -35,4 +36,18 @@ public ShardingSphereUser(final String username, final String password, final St
         grantee = new Grantee(username, hostname);
         this.password = password;
     }
+
+    /**
+     * get decrypt password.
+     * @return String the decrypt password
+     */
+    public String getPassword() {
+        try {
+            return AlgorithmSecureFactory.getInstance().decryptFrontend(password);
+            // CHECKSTYLE:OFF
+        } catch (Exception e) {
+            //checkstyle:ON
+            return null;
```

Comment:
When will the password return null?

Reply:
when password encrypt or decrypt algorithm run failed, then throught exception cause return null.

Reply:
If the NPE throw and log, it will make the end users confused, could you consider about enhance the logic of error handler?

Reply:
already fix.
 I will log.error exception and return empty string. in end user, he will got authenticate failed info.